query/aql_compiler.go

```diff
@@ -766,14 +766,22 @@ func (qc *AQLQueryContext) Rewrite(expression expr.Expr) expr.Expr {
 				e.RHS = expr.Cast(e.RHS, highestType)
 			}
 
-			if rhs != nil && lhs.DataType == memCom.GeoPoint {
+			if rhs != nil && e.LHS.Type() == expr.GeoPoint {
 				if val, err := memCom.GeoPointFromString(rhs.Val); err != nil {
 					qc.Error = err
 				} else {
 					e.RHS = &expr.GeopointLiteral{
 						Val: val,
 					}
 				}
+			} else if rhs != nil && e.LHS.Type() == expr.UUID {
```

Comment:
same here

Reply:
lhs is not always var type, so it could be nil